cpp/src/parquet/encoding.cc

```diff
@@ -4055,4 +4055,57 @@ std::unique_ptr<Decoder> MakeDictDecoder(Type::type type_num,
 }
 
 }  // namespace detail
+
+bool IsParquetVersionAtLeast2_0(ParquetVersion::type parquet_version) {
+  switch (parquet_version) {
+    case ParquetVersion::PARQUET_1_0:
+      return false;
+    case ParquetVersion::PARQUET_2_4:
+    case ParquetVersion::PARQUET_2_6:
+      return true;
+    default:
+      return parquet_version > ParquetVersion::PARQUET_1_0;
+  }
+  return false;
+}
+
+// Informed heavily by https://github.com/apache/parquet-format/blob/master/Encodings.md
+// and
+// https://github.com/apache/arrow-rs/blob/a15adf66a8212fdb3a0222a32130ce70cfc28cf5/parquet/src/column/writer/mod.rs#L1183
+Encoding::type ChooseNonDictEncoding(Type::type data_type,
```

Comment:
We may also need input logical type here because FIXED_LEN_BYTE_ARRAY with logical type FLOAT16|DECIMAL should support BYTE_STREAM_SPLIT in PARQUET_2_11.

cc @pitrou